### PR TITLE
[DCE-156] Walking Skeleton 

### PR DESCRIPTION
**Pull request by DBeniskos**
Create scaffolding for dotnet application

Checklist:
- [x] main branch is protected
- [ ] includes PR template
- [x] github action that writes hello-world using ubunutu runner
- [x] basic dotnet web app that writes "hello, World" do not do any other development
- [x] contains README on to run locally as a ubuntu
- [x] should contains .gitignore

### CODE REVIEW

**Comment by sethpthomas91** [comment]
Can you please add a README with instructions how to run this app?

**Comment by DBeniskos** [comment]
I have updated the README file

File: README.md
Comment (sethpthomas91) [inline README.md:1]: Let's make this even more user friendly. Assume your uer has no idea what requirements your software needs. Think in things like dotnet version. Also let's look at using code snippets in Github markdown.

Some README sources: 
- https://www.makeareadme.com/
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

When I was learning Ruby a few years ago I loved guard's [README](https://github.com/guard/guard#installation)

Comment (DBeniskos) [reply README.md:1]: Ok Im gonna give more instructions

Comment (DBeniskos) [reply README.md:1]: I added more instructions in Readme file. Please have a look.

Comment (sethpthomas91) [reply README.md:1]: In the future please re-request the review on the top right of the conversation page! Thanks!

File: README.md [hunk 1]
```diff
@@ -1 +1,30 @@
...
+You can download a o code editor in order to run this application, you can download VisualStudioCode [here](https://code.visualstudio.com) you have to select the *Linux version*
+#####After installation you have to add some extensions
+1. Open VisualStudioCode, in the left bar click on EXTENSIONS 
+
+2. At the serchbar you can search your extention and click install in order to install it.
+
+Required extensions:
+- c#
+- c# Dev Kit
+- .NET Install Tool
```
Comment (sethpthomas91) [inline README.md:18]:
```suggestion
```

Comment (sethpthomas91) [reply README.md:18]: Good info, but a bit too much. This assumes that folks are using the linux distro. Maybe just scrap this.

File: README.md [hunk 2]
```diff
@@ -1 +1,30 @@
...
+
+### 1. Clone the repository to your local machine
+- go to the folder that you want to clone this repository.
+- open the gitbash inside the folder
+
+- use ```bash git clone git@githubcom:DBeniskos/practice-basic-webapi-application.git``` in order to clone the repository
+### 2. Clone the repository to your local machine
+- open the project folder in your code editor and open terminal
+- in terminal run ```dotnet build``` in order to run this application
```
Comment (sethpthomas91) [inline README.md:27]: `dotnet build` will not give you the "Hello world" output. `dotnet build` only builds the application, but does not run it. `dotnet run --project src/ ` will run the application, but you have to first run a `dotnet build` command beforehand.

**Review by sethpthomas91 — CHANGES_REQUESTED** [review]

**Comment by sethpthomas91** [comment]
Also take a look at the JIRA story and get the rest of this story done please!

**Comment by DBeniskos** [comment]
I have completed the Dotnet Web Application Initial PR pt1

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+jobs:
+  webApi-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Build
+        run: dotnet build ./src/WebApi.csproj
```
Comment (sethpthomas91) [inline .github/workflows/test.yml:17]:
```suggestion
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        run: dotnet test ./test
+
+      - name: Publish
+        run: dotnet publish ./src/WebApi.csproj -c Release -o release — nologo
+
+      - name: Upload Build Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: Application_Artifact
+          path: /home/runner/work/Artifacts/release/
```
Comment (sethpthomas91) [inline .github/workflows/test.yml]:
```suggestion
```
out of scope

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        run: dotnet test ./test
+
+      - name: Publish
+        run: dotnet publish ./src/WebApi.csproj 
+
+      - name: Upload Build Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: Application_Artifact
+          path: /home/runner/work/Artifacts/release/
```
Comment (sethpthomas91) [inline .github/workflows/test.yml:32]:
```suggestion
```
out of scope

File: test/UnitTest1.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+namespace test;
```
Comment (sethpthomas91) [inline test/UnitTest1.cs:1]: The tests were not included as part of DCE-156.

**Review by sethpthomas91 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+name: WebApi
+
+on:    
+  push:
+    branches: 
+      - feature/DCE-156-walking-skeleton
```
Comment (sethpthomas91) [inline .github/workflows/test.yml:6]:
```suggestion
```

Comment (sethpthomas91) [reply .github/workflows/test.yml:6]: We could also look into doing a workflow dispatch in the future, but it is not required now.

File: README.md [hunk 8]
```diff
@@ -1 +1,35 @@
...
+
+### 1. Clone the repository to your local machine
+- go to the folder that you want to clone this repository.
+- open the gitbash inside the folder
+
+- use ```bash git clone git@githubcom:DBeniskos/practice-basic-webapi-application.git``` in order to clone the repository
+### 2. Buld and Run the application
+- open the project folder in your code editor and open terminal
+- in terminal run ```dotnet build``` to build this application
+- after that run ```dotnet run --project src/``` to run the application
```
Comment (sethpthomas91) [inline README.md:27]: Need instructions to shut down. Also Hello world does not display to terminal. It opens up the web api prompt.

**Review by sethpthomas91 — CHANGES_REQUESTED** [review]
Good job. Minor changes requested. Please ask if you have any questions.

**Review by sethpthomas91 — APPROVED** [review]
Great job

**Pull request merged**